mountpoint-s3/tests/common/fuse.rs

```diff
@@ -243,16 +253,14 @@ pub mod mock_session {
     }
 
     impl TestClient for MockTestClient {
-        fn put_object_params(
+        fn put_object_single(
             &self,
             key: &str,
             value: &[u8],
-            params: PutObjectParams,
+            params: PutObjectSingleParams,
         ) -> Result<(), Box<dyn std::error::Error>> {
             let full_key = format!("{}{}", self.prefix, key);
-            let mut mock_object = MockObject::from(value);
-            mock_object.set_storage_class(params.storage_class);
-            self.client.add_object(&full_key, mock_object);
+            _ = self.client.mock_put_object(&full_key, &params, value)?;
```

Comment:
Eliminate the need to export `mock_put_object`?

```suggestion
            _ = tokio_block_on(self.client.put_object_single(BUCKET_NAME, &full_key, &params, value))?;
```